src/pepr/operator/controllers/keycloak/authservice/authservice.ts

```diff
@@ -27,41 +31,82 @@ export const log = setupLogger(Component.OPERATOR_AUTHSERVICE);
 let lock = false;
 
 export async function authservice(pkg: UDSPackage, clients: Map<string, Client>) {
+  if (!pkg.metadata?.namespace || !pkg.metadata?.name) {
+    throw new Error("Package metadata is missing required fields");
+  }
+
   // Get the list of clients from the package
   const authServiceClients = R.filter(
     sso => R.isNotNil(sso.enableAuthserviceSelector),
     pkg.spec?.sso || [],
   );
 
+  // Check if we're in ambient mode by looking for the waypoint annotation
```

Comment:
nit: comment isn't accurate

```suggestion
  // Check if we're in ambient mode by checking the service mesh mode
```

Reply:
done